pygeo/parameterization/DVGeoCST.py

```diff
@@ -164,21 +164,34 @@ def __init__(
         self.foilCoords[:, self.xIdx] = coords[:, 0]
         self.foilCoords[:, self.yIdx] = coords[:, 1]
 
+        # Determine if the dat file is closed (first and last points are the same); remove the duplicate point if so
+        distance = np.linalg.norm(self.foilCoords[0, :] - self.foilCoords[-1, :])
+        distTol = 1e-12
+        if distance < distTol:
+            self.foilCoords = self.foilCoords[:-1, :]
+
+        # Flip the y-coordinates to counter-clockwise if they are clockwise
+        if self.foilCoords[0, self.yIdx] < self.foilCoords[-1, self.yIdx]:
```

Comment:
Does this check break if the airfoil is sharp? Might be better to check indices `1` and `-2` if so.

Reply:
This actually works for sharp airfoils because of the check above it that removes duplicate points

Reply:
Many airfoils (e.g., the [RAE 2822](http://airfoiltools.com/airfoil/details?airfoil=rae2822-il)) have a lower surface slope at the TE that makes the second-to-last point on the lower surface actually has a positive y value. So it might think the lower surface is the upper surface. Would that break this check?

I'm trying to think of all the corner cases and come up with a potentially more robust method. Checking the second to last would work if the x coordinates of all the upper/lower points are the same, but maybe there's some edge case where given different upper and lower x coordinates the y values will make this check wrong. Maybe there's some way to take a vector between two points on each surface at the TE and do some geometry with it? You might be able to use the sign of the Z component of the cross product between the vectors at the TE (idx 1 - 0 and idx -1 - -2) as a check. I haven't thought it through fully, but there's gotta be something simple and robust.

Reply:
Those are both cases where the current check could break. The cross-product approach should work